mhlo_builder/mlir/mhlo/builder/mlir_shape_builder.cpp

```diff
@@ -63,8 +63,8 @@ mlir::Value BuildShapeOfTensor(mlir::OpBuilder& builder,
   auto shape_values = BuildDimSizeListOfTensor(builder, loc, tensor);
   int rank = GetRankOfMlirValue(tensor);
 
-  mlir::Value shape = builder.create<mlir::tensor::FromElementsOp>(
-      loc, BuildMHloDimType(builder), shape_values);
+  mlir::Value shape =
+      builder.create<mlir::tensor::FromElementsOp>(loc, shape_values);
```

Comment:
What's the element type of the result `shape` value? Is that safe to remove `BuildMHloDimType` call?

Reply:
It will by default create a type according to the type of the elements in shape_values.
Since FromElements requires the element type of input/output to be the same. In my understanding, this is safe.